mvlearn/embed/kcca.py

```diff
@@ -147,9 +174,20 @@ def transform(self, Xs):
         if not hasattr(self, "weights_"):
             raise NameError("kCCA has not been trained.")
 
-        self.components_ = _listdot([d.T for d in Xs], self.weights_)
-        self.cancorrs_ = _listcorr(self.components_)
-        self.cancorrs_ = self.cancorrs_[np.nonzero(self.cancorrs_)]
+        weight1 = self.weights_[0]
+        weight2 = self.weights_[1]
+
+        comp1 = []
+        comp2 = []
+
+        for i in range(weight1.shape[1]):
+            comp1.append(self.Kx@weight1[:, i])
+            comp2.append(self.Ky@weight2[:, i])
+
+        comp1 = [l*(-10**18) for l in comp1]
+        comp2 = [l*10**18 for l in comp2]
```

Comment:
make into numpy arrays. components should be list of 2 numpy arrays. As of now its a list of lists of lists

Reply:
also transpose it. X should be (n_samples, n_features) dimension

Reply:
not supposed to transpose it. resulting components are of (n_samples, n_samples) because kcca projects the original (n_sample, n_feature) data to a higher dimensional space

Reply:
I'm just referring to when I call ``components = kcca.transform(Xs)`` I want the returned components to have n rows and n_components feature.

Reply:
oh i see. resolved that